python 3.9.7: failed to build _decimal module

I noticed this while clangarm64 was building python 3.9.7, but looking back at the [autobuild logs](https://github.com/msys2/msys2-autobuild/actions/runs/1260917937) this failed on ucrt64 too at least:
```
  In file included from ../Python-3.9.7/Include/unicodeobject.h:58,
                   from ../Python-3.9.7/Include/Python.h:106,
                   from C:/_/M/mingw-w64-python/src/Python-3.9.7/Modules/_decimal/_decimal.c:29:
  C:/_/M/mingw-w64-python/src/Python-3.9.7/Modules/_decimal/_decimal.c: In function 'dec_format':
  C:/_/M/mingw-w64-python/src/Python-3.9.7/Modules/_decimal/_decimal.c:3282:43: error: 'uchar' undeclared (first use in this function); did you mean 'char'?
   3282 |         if (n > 1 || (n == 1 && !isascii((uchar)spec.dot[0]))) {
        |                                           ^~~~~
```

Python doesn't fail the build if a module fails, it just lacks that module, so this was probably overlooked.

--- a/mingw-w64-python/0116-decimal-backport-build-fixes.patch
+++ b/mingw-w64-python/0116-decimal-backport-build-fixes.patch
@@ -1,0 +1,35 @@
+From f9d77130f2c14f49081ba243f346e0c458f45d5e Mon Sep 17 00:00:00 2001
+From: Christoph Reiter <reiter.christoph@gmail.com>
+Date: Thu, 23 Sep 2021 09:55:32 +0200
+Subject: [PATCH 116/N] decimal: backport build fixes
+
+part of https://github.com/python/cpython/pull/24962
+---
+ Modules/_decimal/_decimal.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/Modules/_decimal/_decimal.c b/Modules/_decimal/_decimal.c
+index d80d791..28fd209 100644
+--- a/Modules/_decimal/_decimal.c
++++ b/Modules/_decimal/_decimal.c
+@@ -3279,7 +3279,7 @@ dec_format(PyObject *dec, PyObject *args)
+     }
+     else {
+         size_t n = strlen(spec.dot);
+-        if (n > 1 || (n == 1 && !isascii((uchar)spec.dot[0]))) {
++        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.dot[0]))) {
+             /* fix locale dependent non-ascii characters */
+             dot = dotsep_as_utf8(spec.dot);
+             if (dot == NULL) {
+@@ -3288,7 +3288,7 @@ dec_format(PyObject *dec, PyObject *args)
+             spec.dot = PyBytes_AS_STRING(dot);
+         }
+         n = strlen(spec.sep);
+-        if (n > 1 || (n == 1 && !isascii((uchar)spec.sep[0]))) {
++        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.sep[0]))) {
+             /* fix locale dependent non-ascii characters */
+             sep = dotsep_as_utf8(spec.sep);
+             if (sep == NULL) {
+-- 
+2.33.0
+

--- a/mingw-w64-python/0117-smoketests-test-that-_decimal-exists.patch
+++ b/mingw-w64-python/0117-smoketests-test-that-_decimal-exists.patch
@@ -1,0 +1,26 @@
+From b7d2e9e27140b56c131e1705bd20dcc7dac63eb2 Mon Sep 17 00:00:00 2001
+From: Christoph Reiter <reiter.christoph@gmail.com>
+Date: Thu, 23 Sep 2021 09:57:56 +0200
+Subject: [PATCH 117/N] smoketests: test that _decimal exists
+
+---
+ mingw_smoketests.py | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/mingw_smoketests.py b/mingw_smoketests.py
+index 70acbd1..527aafb 100644
+--- a/mingw_smoketests.py
++++ b/mingw_smoketests.py
+@@ -139,6 +139,9 @@ class Tests(unittest.TestCase):
+         import ctypes
+         import curses
+ 
++    def test_c_modules_import(self):
++        import _decimal
++
+     def test_socket_inet_ntop(self):
+         import socket
+         self.assertTrue(hasattr(socket, "inet_ntop"))
+-- 
+2.33.0
+

--- a/mingw-w64-python/PKGBUILD
+++ b/mingw-w64-python/PKGBUILD
@@ -22,7 +22,7 @@ else
   pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}${_pybasever}")
 fi
 pkgver=${_pybasever}.7
-pkgrel=1
+pkgrel=2
 pkgdesc="A high-level scripting language (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -162,7 +162,9 @@ source=("https://www.python.org/ftp/python/${pkgver%rc?}/Python-${pkgver}.tar.xz
         0112-expanduser-normpath-paths-coming-from-env-vars.patch
         0113-Add-support-for-Windows-7-Python-3.9.patch
         0114-Commit-regenerated-importlib.patch
-        0115-CI-test-the-build-and-add-some-mingw-specific-tests.patch)
+        0115-CI-test-the-build-and-add-some-mingw-specific-tests.patch
+        0116-decimal-backport-build-fixes.patch
+        0117-smoketests-test-that-_decimal-exists.patch)
 
 # Helper macros to help make tasks easier #
 apply_patch_with_msg() {
@@ -290,7 +292,9 @@ prepare() {
   0112-expanduser-normpath-paths-coming-from-env-vars.patch \
   0113-Add-support-for-Windows-7-Python-3.9.patch \
   0114-Commit-regenerated-importlib.patch \
-  0115-CI-test-the-build-and-add-some-mingw-specific-tests.patch
+  0115-CI-test-the-build-and-add-some-mingw-specific-tests.patch \
+  0116-decimal-backport-build-fixes.patch \
+  0117-smoketests-test-that-_decimal-exists.patch
  
   autoreconf -vfi
 }
@@ -501,4 +505,6 @@ sha256sums=('f8145616e68c00041d1a6399b76387390388f8359581abc24432bb969b5e3c57'
             'c4e3e7903b17056a97af98a7b1e9737b323e51097e967789d3a2796914a8d6af'
             'a0c51295efba6111c9916d26decf810119f56695bdb2776179367ce291a875d4'
             '14251dcc7b776b2640941f675be4449addb598537b522c4e574f9a4bc37c65c1'
-            'e4d481bc25f6386c6da99e2cbe058b21bc0c68bf8f01c73d3adf60e98f5a506d')
+            'e4d481bc25f6386c6da99e2cbe058b21bc0c68bf8f01c73d3adf60e98f5a506d'
+            '0e0766e3aece0a9cfa8e4e0e6592d6f5082ffc686b6b195a3fdb942b5b24ec29'
+            '169b29cb4eb9177938d66b99234a83d5efcc5b0b759d4e1b9ef2acb357e11a15')

